probatus/utils/shap_helpers.py

```diff
@@ -92,7 +92,17 @@ def shap_calc(
             pass
         mask = shap.utils.sample(X, sample_size)
 
-        explainer = shap.Explainer(model, masker=mask, **shap_kwargs)
+        # For LGBM to work with categorical features, we need to exclude background data from shap
+        model_str = str(model) if not isinstance(model, Pipeline) else str(model[-1])
+        if (X.select_dtypes("category").shape[1] > 0) and model_str.startswith("LGBM"):
```

Comment:
XGBoost does not accept categorical input, only numerical. Couldn't find an official link that states this clearly, but there are plenty of resources about this.
catboost I've never used, but it seems they can use categorical input.

I'm not sure what other alternatives there would be apart from importing all models and checking if the types are the same (which we shouldn't do lest you want to add dependencies for all these libraries)

Reply:
Just checking how `shap` does it - they have a utility they call `safe_isinstance` we may be able to use (https://github.com/slundberg/shap/blob/master/shap/utils/_general.py). However, 1. it also wouldn't work with a wrapper class, 2. since `shap` will do this anyways internally, I think a string check is fine for this purpose.

Reply:
I think it is a bit unsafe to use the internal function from shap. For now we can stick to the string check, and if it causes issues to users in the future, we can remove it afterwards.

i also think it could be fine to not check the class name at all, since there are no other models that we could explain with categorical features, since we do not support model pipelines.